atef/procedure.py

```diff
@@ -216,27 +242,41 @@ class CodeStep(ProcedureStep):
 @dataclass
 class PlanOptions:
     """Options for a bluesky plan scan."""
-    #: The plan name.
+    #: Name to identify this plan
+    name: str
+    #: The plan name.  Bluesky plan or otherwise
     plan: str
     #: Plan arguments dictionary - argument name to value.
-    args: Sequence[Any]
+    args: Optional[Sequence[Any]] = field(default_factory=list)
     #: Plan keyword  arguments dictionary - argument name to value.
-    kwargs: Dict[Any, Any]
+    kwargs: Optional[Dict[Any, Any]] = field(default_factory=dict)
     #: Arguments which should not be configurable.
-    fixed_arguments: Optional[Sequence[str]]
+    fixed_arguments: Optional[Sequence[str]] = None
+
+    def make_plan_item(self: PlanOptions) -> Dict[str, Any]:
```

Comment:
Nit - many similar functions in dataclasses are named `from_X` or `to_Y` in this codebase